mono/metadata/class.c

```diff
@@ -1889,6 +1889,10 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 
 		if (!(field->type->attrs & FIELD_ATTRIBUTE_STATIC)) {
 			ftype = mono_type_get_underlying_type (field->type);
+			if (!ftype) {
+				mono_class_set_type_load_failure (klass, "Field '%s' is an enum type with a bad underlying type", field->name);
+				continue;
+			}
```

Comment:
Windows gets a TLE on `E` at the `var t = tb.CreateType()`. (With my patch, we get a TLE on `T` with a message about its `e` field at `Activator.CreateInstance (t)`.)

windows error:
`System.TypeLoadException occurred
  HResult=0x80131522
  Message=Could not load type 'E' from assembly 'Thing, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.`